Apollo/Binary/Encoder.cs

```diff
@@ -274,6 +274,13 @@ public static class Encoder {
             writer.Write(o.Infinite);
             writer.Write(o.Release);
         }
+        
+        static void Encode(BinaryWriter writer, Loop o){
+            EncodeID(writer, typeof(Loop));
```

Comment:
This will crash as you've not added Loop to Common.cs where the IDs for each object are listed.